zipkin-server/src/main/java/zipkin2/server/internal/ZipkinServerConfiguration.java

```diff
@@ -86,8 +87,15 @@ public void addViewControllers(ViewControllerRegistry registry) {
   /** Configures the server at the last because of the specified {@link Order} annotation. */
   @Order @Bean ArmeriaServerConfigurator corsConfigurator(
     @Value("${zipkin.query.allowed-origins:*}") String allowedOrigins) {
-    CorsServiceBuilder corsBuilder =
-      CorsServiceBuilder.forOrigins(allowedOrigins.split(",")).allowRequestMethods(HttpMethod.GET);
+    CorsServiceBuilder corsBuilder = CorsServiceBuilder.forOrigins(allowedOrigins.split(","))
+      // NOTE: The property says query, and the UI does not use POST, but we allow POST?
+      //
+      // The reason is that our former CORS implementation accidentally allowed POST. People doing
```

Comment:
@drolando this is one of the gotchas that probably works in your favor when thinking about having the UI post-back. I didn't realize we accidentally allowed POST anyway!